drivers/dome/scopedome_usb21.cpp

```diff
@@ -326,12 +327,11 @@ int ScopeDomeUSB21::readBuf(Command &cmd, uint8_t len, uint8_t *buff)
 {
     int nbytes_read = 0, rc = -1;
     int BytesToRead = len + 4;
-    uint8_t cbuf[BytesToRead];
-    memset(cbuf, 0, BytesToRead);
+    std::vector<uint8_t> cbuf(BytesToRead, 0);
     char errstr[MAXRBUF];
 
     // Read buffer
-    if ((rc = tty_read(PortFD, (char *)cbuf, sizeof(cbuf), SCOPEDOME_TIMEOUT, &nbytes_read)) != TTY_OK)
+    if ((rc = tty_read(PortFD, (char *)cbuf.data(), sizeof(cbuf), SCOPEDOME_TIMEOUT, &nbytes_read)) != TTY_OK)
```

Comment:
Are you sure this can work?? I don't think sizeof would work here. Maybe cbuf.size

Also, if we use this:

```uint8_t cbuf[BytesToWrite] = {0};```

Wouldn't that be sufficient?

Reply:
You're right, I didn't check the sizeof calculation. Let me update that
The problem is that `BytesToWrite` isn't known at compile-time. So `uint8_t cbuf[BytesToWrite] = {0};` isn't legal c++ still makes use of the Clang extension.

Reply:
updated to return the calculation of the size in bytes